samples/subgroup-scan.cpp

```diff
@@ -0,0 +1,208 @@
+/***************************************************************************
+ *
+ *  Copyright (C) 2017 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's ComputeCpp SDK
+ *
+ *  scan-subgroups.cpp
+ *
+ *  Description:
+ *    Example of a parallel inclusive scan in SYCL using subgroup operations.
+ *
+ **************************************************************************/
+
+#include <CL/sycl.hpp>
+namespace sycl = cl::sycl;
+
+#include <algorithm>
+#include <iostream>
+#include <numeric>
+#include <vector>
+
+// Dummy struct to generate unique kernel name types
+template <typename T, typename U, typename V>
+struct kernel_name {};
+
+/* Performs an inclusive scan with the given associative binary operation `Op`
+ * on the data in the `in` buffer. Runs in parallel on the provided accelerated
+ * hardware queue. Modifies the input buffer to contain the results of the scan.
+ * Input size has to be a power of two. If the size isn't so, the input can
+ * easily be padded to the nearest power of two with any values, and the scan on
+ * the meaningful part of the data will stay the same. */
+template <typename T, typename Op>
+void par_scan(sycl::buffer<T, 1> in, sycl::queue q) {
+  if ((in.get_count() & (in.get_count() - 1)) != 0 || in.get_count() == 0) {
```

Comment:
`get_count()` is deprecated. Use `size()` instead. https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_buffer_interface table 39.

Reply:
Has anyone made a bitesize joke yet?